### PR TITLE
Image enhancements

### LINKED ISSUE #1197: Client Side Image Enhancement
TODO fill this out.

### PR DESCRIPTION
**Pull request by BryonLewis**
fixes #1197 

I added in a small basic image enhancement to modify the filter on SVG filter.

I added a context menu because I have a feeling these effects might expand in the future beyond just doing contrast adjustments.  I guess it could be a small pop-up like the playback speed, but I'm guessing that we want to add more stuff in the future to this.

Image/Video Annotators have been modified to accepting an [intercept](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/intercept) as well as a brightness setting.  So you can adjust both to adjust the brightness and contrast of the image.

I created `useImageEnhancments.ts` with the idea that in the future the image enhancements may be more complex with more data being transferred.  That's also why it accepts `markChangesPending`, so that the setting could be saved in the future for the dataset metadata.

image

### CODE REVIEW

File: client/src/use/useImageEnhancements.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+
+export default function useImageEnhancements({ markChangesPending }: ImageEnhancementParams) {
+  const imageEnhancements: Ref<ImageEnhancements> = ref({});
+
+  const setSVGFilters = ({ blackPoint, whitePoint }:
+    {blackPoint?: number; whitePoint?: number }) => {
+    VueSet(imageEnhancements.value, 'blackPoint', blackPoint);
+    VueSet(imageEnhancements.value, 'whitePoint', whitePoint);
+    markChangesPending();
```
Comment (subdavis) [inline client/src/use/useImageEnhancements.ts:29]: This composition function should not call markChangesPending.  There's nothing to save.  It should probably not take it as a param.

**Review by subdavis — CHANGES_REQUESTED** [review]
Works perfectly, code looks great.

Just need to remove `markChangesPending`

Edit: can you also add this line to documentation in `UI-Timeline.md`

```
* ==:material-contrast-box:== will open the image contrast adjustment dialog.
```

File: client/src/use/useImageEnhancements.ts (hunk 1 above)
Comment (BryonLewis) [reply client/src/use/useImageEnhancements.ts:29]: :facepalm: -  I was in the middle of adding saving of the metadata, and I must have forgot to remove it.

**Review by subdavis — APPROVED** [review]
:100:

**Pull request merged**